lib/galaxy/jobs/__init__.py

```diff
@@ -1809,27 +1809,11 @@ def requires_setting_metadata( self ):
             return self.tool.requires_setting_metadata
         return False
 
-    def _report_error_to_sentry( self ):
+    def _report_error( self ):
         job = self.get_job()
         tool = self.app.toolbox.get_tool(job.tool_id, tool_version=job.tool_version) or None
-        if self.app.sentry_client and job.state == job.states.ERROR:
-            self.app.sentry_client.capture(
-                'raven.events.Message',
-                message="Galaxy Job Error: %s  v.%s" % (job.tool_id, job.tool_version),
-                extra={
-                    'info' : job.info,
-                    'id' : job.id,
-                    'command_line' : job.command_line,
-                    'stderr' : job.stderr,
-                    'traceback': job.traceback,
-                    'exit_code': job.exit_code,
-                    'stdout': job.stdout,
-                    'handler': job.handler,
-                    'user': self.user,
-                    'tool_version': job.tool_version,
-                    'tool_xml': tool.config_file if tool else None
-                }
-            )
+        for dataset in job.output_datasets:
+            self.app.error_reports.default_error_sink.submit_report(dataset, job, tool, user_submission=False)
```

Comment:
@jmchilton is this the right way to do this? I get the impression that it isn't, but I wasn't understanding the lambda expression that I copied from your metrics. I.e. https://github.com/galaxyproject/galaxy/pull/4305/files#diff-218b8eec0cd1aaab08f26181429f4ba0R19